subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ComponentVariantIdentifier.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.attributes.AttributeContainer;
+import org.gradle.api.capabilities.Capability;
+
+import java.util.List;
+
+/**
+ * Identifier of a variant of a component.
+ */
+public class ComponentVariantIdentifier {
```

Comment:
🤔 I wonder if we can get away with only having one type for `ComponentVariantIdentifier`, or maybe two different types with different names.

I see that we use Gradle's domain types in `TransformedProjectArtifactSet`, though not further down. Maybe it would make sense to pass the domain types in there and then only have the build operation type in the TransformationNodes. We would then also store the build operation types in the configuration cache. WDYT?